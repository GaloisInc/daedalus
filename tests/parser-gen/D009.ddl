-- Testing stream operations with some static data-dependency
-- should be same output as D008.ddl

def Chunk n P =  {
  @cur  = GetStream;
  @this = Take n cur;
  @next = Drop n cur;
  SetStream this;
  $$ = P;
  SetStream next;
}

def PadWSpaces n P = 
  Chunk n {$$ = P; Many (Match1 ' '); END}

def Main =
  { @a = Choose
    { x = PadWSpaces 3 { Match "ab"}
    ; y = PadWSpaces 3 { Match "ac"}
    }
  ; END
  }
