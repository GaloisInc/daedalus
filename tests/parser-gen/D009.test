dump
D009.ddl
--gen
