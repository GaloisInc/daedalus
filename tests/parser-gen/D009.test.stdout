Stack:
    *
    Main callsite __START__
    _PadWSpaces__30 callsite Main (18,11)-(18,36)
    _Chunk__35 callsite _PadWSpaces__30 (14,3)-(14,42)
    Many
Start: _Chunk__35 (14,20)-(14,35)
SlkCfg{ q:38, ctrl:[?, ?, ?, ?, *], sem:[?, ?, ?, ], inp:---[2..3] }
DTrans [
  ( [(7),]
  , HeadInput [' ']
  , Resolution (NotAmbiguous)
  ),
  ( []
  , EndInput
  , Resolution (NotAmbiguous)
  ),
]

Stack:
    *
    Main callsite __START__
    _PadWSpaces__30 callsite Main (18,11)-(18,36)
    _Chunk__35 callsite _PadWSpaces__30 (14,3)-(14,42)
    Many
Start: _Chunk__35 (14,20)-(14,35)
SlkCfg{ q:38, ctrl:[?, ?, ?, ?, *], sem:[?, ?, ?, ], inp:---[3..3] }
DTrans [
  ( []
  , HeadInput [' ']
  , Resolution (NotAmbiguous)
  ),
  ( [(33),]
  , EndInput
  , Resolution (NotAmbiguous)
  ),
]

Stack:
    *
    Main callsite __START__
    _PadWSpaces__31 callsite Main (19,11)-(19,36)
    _Chunk__34 callsite _PadWSpaces__31 (14,3)-(14,42)
    Many
Start: _Chunk__34 (14,20)-(14,35)
SlkCfg{ q:110, ctrl:[?, ?, ?, ?, *], sem:[?, ?, ?, ], inp:---[2..3] }
DTrans [
  ( [(7),]
  , HeadInput [' ']
  , Resolution (NotAmbiguous)
  ),
  ( []
  , EndInput
  , Resolution (NotAmbiguous)
  ),
]

Stack:
    *
    Main callsite __START__
    _PadWSpaces__31 callsite Main (19,11)-(19,36)
    _Chunk__34 callsite _PadWSpaces__31 (14,3)-(14,42)
    Many
Start: _Chunk__34 (14,20)-(14,35)
SlkCfg{ q:110, ctrl:[?, ?, ?, ?, *], sem:[?, ?, ?, ], inp:---[3..3] }
DTrans [
  ( []
  , HeadInput [' ']
  , Resolution (NotAmbiguous)
  ),
  ( [(33),]
  , EndInput
  , Resolution (NotAmbiguous)
  ),
]

Stack:
    *
    Main callsite __START__
Start: Main (21,5)-(21,7)
SlkCfg{ q:166, ctrl:[?, *], sem:[?, ], inp:---[3..] }
DTrans [
  ( [(9),]
  , EndInput
  , Resolution (NotAmbiguous)
  ),
]

Stack:
    *
Start: Main (16,1)-(22,3)
SlkCfg{ q:1, ctrl:[*], sem:[*], inp:--- }
DTrans [
  ( [(40),(40),]
  , HeadInput ['a']
  , DTrans [
      ( [(9),]
      , HeadInput ['b']
      , Resolution (NotAmbiguous)
      ),
      ( [(9),]
      , HeadInput ['c']
      , Resolution (NotAmbiguous)
      ),
    ]
  ),
]

Stack:
    *
Start: __START__
SlkCfg{ q:0, ctrl:[*], sem:[?, *], inp:--- }
DTrans [
  ( [(0),]
  , AcceptingPath
  , NotAmbiguous
  ),
]

__ZSINK_STATE


**********************
***** Strict LLA *****
**********************
("Lookahead-0",2)
("Lookahead-1",5)
("Lookahead-2",1)

Total nb DFAs: 8

SUCCESS: Fully determinized format
